flatfs.go

```diff
@@ -55,6 +56,10 @@ var (
 	ErrShardingFileMissing   = fmt.Errorf("%s file not found in datastore", SHARDING_FN)
 )
 
+func init() {
+	rand.Seed(time.Now().UTC().UnixNano())
```

Comment:
I would switch to package local `Rand` instance as you can't guarantee that `Seed` won't be called with other calls to `Rand` concurrently.

Reply:
I don' t understand what you mean

Reply:
`Seed, unlike the Rand.Seed method, is safe for concurrent use. ` https://golang.org/pkg/math/rand/#Seed

Reply:
I've missed it, sorry. I was looking at Rand.Seed.